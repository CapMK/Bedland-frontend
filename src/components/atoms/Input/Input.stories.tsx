import type { Meta, StoryObj } from '@storybook/react';
import { useForm, FieldValues, UseFormRegister } from 'react-hook-form';
import ThemeProvider from 'theme/ThemeContext';
import GlobalStyle from 'theme/globalStyles';
import loginRegexPattern from 'utils/loginRegexPatterns';
import Input from './Input';

const meta = {
  title: 'Atoms/Input',
  component: Input,
  tags: ['autodocs'],
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        {Story()}
      </ThemeProvider>
    ),
  ],
} satisfies Meta<typeof Input>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  render: (args) => {
    const { register } = useForm();
    return (
      <form>
        <Input {...args} register={register} />
      </form>
    );
  },
  args: {
    input: 'login',
    type: 'text',
    placeholder: 'Your login',
    register: {} as UseFormRegister<FieldValues>,
    regexPattern: loginRegexPattern.login,
  },
};
