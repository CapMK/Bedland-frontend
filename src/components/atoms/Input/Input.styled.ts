import styled from 'styled-components';
import { LOGIN_INPUT_WIDTH } from 'utils/constans';

export const LoginInput = styled.input<{ borderColor: string }>`
  border: 1px solid ${(props) => props.borderColor};
  border-radius: 0.5rem;
  display: flex;
  height: 2.875rem;
  padding: 0.625rem 1rem 0.625rem 1rem;
  width: ${LOGIN_INPUT_WIDTH}rem;
  &:disabled,
  &:hover,
  &:active,
  &:focus {
    outline: none;
  }
`;

export default LoginInput;
