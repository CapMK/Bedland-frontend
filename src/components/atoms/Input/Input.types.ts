import { FieldValues, UseFormRegister } from 'react-hook-form';
import { HTMLInputTypeAttribute } from 'react';

export type TRegister = {
  pattern?: RegExp;
  required: boolean;
  maxLength?: number;
  minLength?: number;
};

const inputVariants = ['login', 'password', 'confirmPassword'] as const;

export type InputVariant = (typeof inputVariants)[number];

export type RegisterOptions = { validate: (value: string) => boolean } | null;

export type InputProps = {
  input: InputVariant;
  type: HTMLInputTypeAttribute;
  placeholder: string;
  register: UseFormRegister<FieldValues>;
  registerOptions?: RegisterOptions;
  regexPattern: TRegister;
};
