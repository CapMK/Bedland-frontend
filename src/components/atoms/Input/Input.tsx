import { useContext } from 'react';
import { ThemeContext } from 'theme/ThemeContext';
import * as Styled from './Input.styled';
import { InputProps } from './Input.types';

const Input = ({
  input,
  type = 'text',
  placeholder,
  register,
  registerOptions = null,
  regexPattern,
}: InputProps) => {
  const { palette } = useContext(ThemeContext);

  return (
    <Styled.LoginInput
      id={input}
      type={type}
      placeholder={placeholder}
      borderColor={palette.sidebarTabsPressed}
      data-testid={input}
      {...register(input, { ...regexPattern, ...registerOptions })}
    />
  );
};

export default Input;
