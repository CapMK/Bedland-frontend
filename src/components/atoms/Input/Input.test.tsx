import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { useForm } from 'react-hook-form';
import ThemeProvider from 'theme/ThemeContext';
import loginRegexPattern from 'utils/loginRegexPatterns';
import Input from './Input';

const SnapshotInput = () => {
  const { register } = useForm();
  return (
    <ThemeProvider>
      <form action=''>
        <Input
          input='login'
          type='text'
          placeholder='login'
          register={register}
          regexPattern={loginRegexPattern.login}
        />
      </form>
    </ThemeProvider>
  );
};

describe('Input', () => {
  afterEach(cleanup);
  it('renders correctly', () => {
    const snapshot = renderer.create(<SnapshotInput />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
