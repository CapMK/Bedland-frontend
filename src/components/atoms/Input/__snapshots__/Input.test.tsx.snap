// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input > renders correctly 1`] = `
.c0 {
  border: 1px solid #919AFF;
  border-radius: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2.875rem;
  padding: 0.625rem 1rem 0.625rem 1rem;
  width: 19.875rem;
}

.c0:disabled,
.c0:hover,
.c0:active,
.c0:focus {
  outline: none;
}

<form
  action=""
>
  <input
    className="c0"
    data-testid="login"
    id="login"
    name="login"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="login"
    type="text"
  />
</form>
`;
